recipes/TIMIT/ASR/transformer/hparams/transformer_madgrad.yaml

```diff
@@ -0,0 +1,202 @@
+
+# Data files
+data_folder: !PLACEHOLDER  # e.g. /path/to/TIMIT
+train_annotation: !ref <data_folder>/train.csv
+valid_annotation: !ref <data_folder>/dev.csv
+test_annotation: !ref <data_folder>/test.csv
+skip_prep: False # Skip data preparation
+
+# Training parameters
+number_of_epochs: 140
+batch_size: 16
+ctc_weight: 0.3
+gradient_accumulation: 4
+gradient_clipping: 5.0
+loss_reduction: 'batchmean'
+sorting: random
+emb_size: 128
+
+# stages related parameters
+stage_one_epochs: 118
+lr_noam: 2.0
+lr_sgd: 0.000025
+
+# Feature parameters
+sample_rate: 16000
+n_fft: 400
+n_mels: 40
+
+# Dataloader options
+train_dataloader_opts:
+    batch_size: !ref <batch_size>
+    shuffle: True
+
+valid_dataloader_opts:
+    batch_size: 1
+
+test_dataloader_opts:
+    batch_size: 1
+
+####################### Model parameters ###########################
+# Transformer
+d_model: 512
+nhead: 8
+num_encoder_layers: 6
+num_decoder_layers: 6
+d_ffn: 3072
+transformer_dropout: 0.2
+activation: !name:torch.nn.GELU
+output_neurons: 42
+vocab_size: 42
+dec_neurons: 256
+dnn_neurons: 512
+
+# Outputs
+blank_index: 0
+label_smoothing: 0.1
+pad_index: 0
+bos_index: 1
+eos_index: 2
+unk_index: 0
+
+# Decoding parameters
+min_decode_ratio: 0.0
+max_decode_ratio: 1.0
+beam_size: 16
+valid_search_interval: 10
+valid_beam_size: 10
+test_beam_size: 66
+ctc_weight_decode: 0.52
+
+emb: !new:speechbrain.nnet.embedding.Embedding
+    num_embeddings: !ref <output_neurons>
+    embedding_dim: !ref <emb_size>
+
+############################## models ################################
+
+CNN: !new:speechbrain.lobes.models.convolution.ConvolutionFrontEnd
+    input_shape: (8, 10, 40)
+    num_blocks: 3
+    num_layers_per_block: 1
+    out_channels: (128, 256, 512)
+    kernel_sizes: (3, 3, 1)
+    strides: (2, 2, 1)
+    residuals: (False, False, False)
+
+Transformer: !new:speechbrain.lobes.models.transformer.TransformerASR.TransformerASR # yamllint disable-line rule:line-length
+    input_size: 5120
+    tgt_vocab: !ref <output_neurons>
+    d_model: !ref <d_model>
+    nhead: !ref <nhead>
+    num_encoder_layers: !ref <num_encoder_layers>
+    num_decoder_layers: !ref <num_decoder_layers>
+    d_ffn: !ref <d_ffn>
+    dropout: !ref <transformer_dropout>
+    activation: !ref <activation>
+    normalize_before: False
+
+
+ctc_lin: !new:speechbrain.nnet.linear.Linear
+    input_size: !ref <d_model>
+    n_neurons: !ref <output_neurons>
+
+seq_lin: !new:speechbrain.nnet.linear.Linear
+    input_size: !ref <d_model>
+    n_neurons: !ref <output_neurons>
+
+modules:
+    CNN: !ref <CNN>
+    Transformer: !ref <Transformer>
+    seq_lin: !ref <seq_lin>
+    ctc_lin: !ref <ctc_lin>
+
+model: !new:torch.nn.ModuleList
+    - [!ref <CNN>, !ref <Transformer>, !ref <seq_lin>, !ref <ctc_lin>]
+
+# define two optimizers here for two-stage training
+# Reference for a good LR - https://github.com/facebookresearch/madgrad (might need clean sweep)
+optim: !name:madgrad.MADGRAD
+    lr: 2.54e-4
```

Comment:
@mravanelli I did some tests, and this line does indeed affect the LR and convergence. 2.54e-4 seems to be the best value for TIMIT after our tests